src/js/ripple/amount.js

```diff
@@ -453,56 +322,37 @@ Amount.prototype.invert = function() {
  * @returns {Amount}
  * @throws {Error} if offset exceeds legal ranges, meaning the amount value is bigger than supported
  */
-Amount.prototype.canonicalize = function() {
-  if (!(this._value instanceof BigInteger)) {
-    // NaN.
-    // nothing
-  } else if (this._is_native) {
-    // Native.
-    if (this._value.equals(BigInteger.ZERO)) {
-      this._offset      = 0;
-      this._is_negative = false;
-    } else {
-      // Normalize _offset to 0.
-
-      while (this._offset < 0) {
-        this._value  = this._value.divide(Amount.bi_10);
-        this._offset += 1;
-      }
-
-      while (this._offset > 0) {
-        this._value  = this._value.multiply(Amount.bi_10);
-        this._offset -= 1;
-      }
-    }
 
-  } else if (this.is_zero()) {
-    this._offset      = Amount.cMinOffset;
-    this._is_negative = false;
+Amount.prototype.canonicalize = function(roundingMode) {
+  if (this._is_native) {
+    var value = this._value.round(6, BigNumber.ROUND_DOWN);
+    this._value = value.absoluteValue().greaterThan(Amount.bi_xrp_max)
```

Comment:
While historically it seems to_text() would return '0' for native amounts that overflow, internally it sets NaN as the value_. Elsewhere we blow up when limits are exceeded. There's no comments here explaining this strange behaviour or what a future course of action to fix it might be. What are the plans for fixing up amount.js beyond using bignumber.js (which by itself doesn't really buy much)? 

In any case, this seems to be a behavioural change.